packages/@aws-cdk/aws-ivs-alpha/lib/recording-configuration.ts

```diff
@@ -0,0 +1,366 @@
+import { CfnRecordingConfiguration } from 'aws-cdk-lib/aws-ivs';
+import { IBucket } from 'aws-cdk-lib/aws-s3';
+import { Duration, Fn, IResource, Resource, Stack, Token } from 'aws-cdk-lib/core';
+import { Construct } from 'constructs';
+
+/**
+ * Properties of the IVS Recording configuration
+ */
+export interface RecordingConfigurationProps {
+  /**
+   * S3 bucket where recorded videos will be stored.
+   */
+  readonly bucket: IBucket;
+
+  /**
+   * The name of the Recording configuration.
+   * The value does not need to be unique.
+   *
+   * @default - auto generate
+   */
+  readonly recordingConfigurationName?: string;
+
+  /**
+   * If a broadcast disconnects and then reconnects within the specified interval,
+   * the multiple streams will be considered a single broadcast and merged together.
+   *
+   * `recordingReconnectWindow` must be between 0 and 300 seconds
+   *
+   * @default - 0 seconds (means disabled)
+   */
+  readonly recordingReconnectWindow?: Duration;
+
+  /**
+   * The set of renditions are recorded for a stream.
+   * For BASIC channels, the CUSTOM value has no effect.
+   * Custom renditions do not apply to BASIC channels.
+   *
+   * @default RenditionSelection.ALL
+   */
+  readonly renditionSelection?: RenditionSelection;
+
+  /**
+   * A list of which renditions are recorded for a stream.
+   * This property must be set when `renditionSelection` is `RenditionSelection.CUSTOM`.
+
+   * @default - no resolution selected
+   */
+  readonly renditions?: Resolution[];
+
+  /**
+   * Thumbnail recording mode.
+   *
+   * @default ThumbnailRecordingMode.INTERVAL
+   */
+  readonly thumbnailRecordingMode?: ThumbnailRecordingMode;
```

Comment:
Would also prefer to use an enum-like class here to enforce which properties must be set together and make it impossible to set conflicting properties together. Then we could remove the `validateThumbnailSettings` method.